flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/constraints/AbstractConstraint.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api.constraints;
+
+import org.apache.flink.annotation.Internal;
+
+import java.util.Objects;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Base class for {@link Constraint constraints}.
+ */
+@Internal
+abstract class AbstractConstraint implements Constraint {
+	private final String name;
+	private final boolean enforced;
+
+	AbstractConstraint(String name, boolean enforced) {
+		this.name = checkNotNull(name);
+		this.enforced = enforced;
+	}
+
+	@Override
+	public String getName() {
+		return name;
+	}
+
+	@Override
+	public boolean isEnforced() {
+		return enforced;
+	}
+
+	/**
+	 * Returns the in brackets part of a constraint summary.
+	 */
+	abstract String constraintDefinition();
+
+	/**
+	 * Returns constraint's summary. All constraints summary will be formatted as
+	 * <pre>
+	 * CONSTRAINT [constraint-name] [constraint-type] ([constraint-definition])
```

Comment:
how about we leave `constraint-type` information also into the responsibility of `constraintDefinition`? It will be easier for us to extend when introducing other constraints which might not be able to follow the pattern `type (def)`.

Reply:
I mean I could move the whole implementation of `asSummaryString` to the leaf class (if we move the constraint-type down, there is not much left here). We can do that at any point in time though. This is an internal implementation. IMHO the next and only constraint that we will potentially add is the `CHECK` constraint, which actually also follows this pattern.

I'm fine with moving it though. WDYT?

Reply:
any chance we would support referential constraint in the future? If yes, I would suggest to change this since it doesn't follow this pattern.

Reply:
I think we could, but you're right. Let's try to unify it once we have more constraints in place. I will move the logic to `UniqueConstraint`.